contracts/src/v0.8/llo-feeds/RewardManager.sol

```diff
@@ -0,0 +1,311 @@
+  event FeePaid(bytes32 poolId, address payee, uint256 quantity);
+
+  /**
+   * @notice Constructor
+   * @param linkAddress address of the wrapped LINK token
+   */
+  constructor(address linkAddress) ConfirmedOwner(msg.sender) {
+    //ensure that the address ia not zero
+    if (linkAddress == address(0)) revert InvalidAddress();
+
+    i_linkAddress = linkAddress;
+  }
+
+  // @inheritdoc TypeAndVersionInterface
+  function typeAndVersion() external pure override returns (string memory) {
+    return "RewardManager 0.0.1";
+  }
+
+  // @inheritdoc IERC165
+  function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
+    return interfaceId == this.onFeePaid.selector;
+  }
+
+  modifier onlyOwnerOrFeeManager() {
+    if (msg.sender != owner() && msg.sender != s_feeManagerAddress) revert Unauthorized();
+    _;
+  }
+
+  modifier onlyOwnerOrRecipientInPool(bytes32 poolId) {
+    if (msg.sender != owner() && s_rewardRecipientWeights[poolId][msg.sender] == 0) revert Unauthorized();
+    _;
+  }
+
+  /// @inheritdoc IRewardManager
+  function onFeePaid(bytes32 poolId, address payee, Common.Asset calldata fee) external override {
+    //this pool only handles LINK fees
+    if (fee.assetAddress != i_linkAddress) {
+      revert InvalidAddress();
+    }
+
+    //update the total fees collected for this pot
+    unchecked {
+      //the total amount for any ERC20 asset cannot exceed 2^256 - 1
+      s_totalRewardRecipientFees[poolId] += fee.amount;
+    }
+
+    //transfer the fee to this contract
+    IERC20(i_linkAddress).transferFrom(payee, address(this), fee.amount);
+
+    emit FeePaid(poolId, payee, fee.amount);
+  }
+
+  /// @inheritdoc IRewardManager
+  function claimRewards(bytes32[] memory poolIds) external override {
+    _claimRewards(msg.sender, poolIds);
+  }
+
+  // wrapper impl for claimRewards
+  function _claimRewards(address recipient, bytes32[] memory poolIds) internal returns (uint256) {
+    //get the total amount claimable for this recipient
+    uint256 claimAmount;
+
+    //loop and claim all the rewards in the poolId pot
+    for (uint256 i; i < poolIds.length; ++i) {
+      //get the poolId to be claimed
+      bytes32 poolId = poolIds[i];
+
+      //get the total fees for the pot
+      uint256 totalFeesInPot = s_totalRewardRecipientFees[poolId];
+
+      unchecked {
+        //get the claimable amount for this recipient, this calculation will never exceed the amount in the pot
+        uint256 claimableAmount = totalFeesInPot - s_totalRewardRecipientFeesLastClaimedAmounts[poolId][recipient];
+
+        //calculate the recipients share of the fees, which is their weighted share of the difference between the last amount they claimed and the current amount in the pot. This can never be more than the total amount in existence
+        uint256 recipientShare = (claimableAmount * s_rewardRecipientWeights[poolId][recipient]) / PERCENTAGE_SCALAR;
```

Comment:
Can `claimableAmount * s_rewardRecipientWeights[poolId][recipient]` be less than `PERCENTAGE_SCALAR;`? Is it ok if `recipientShare` rounds down to 0?